File: MergeSortedArray/MergeSortedArray.cpp
#include <vector>
#include <algorithm> // For std::sort
using namespace std;

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        // Copy nums2 to the end of nums1
        for (int i = 0; i < n; ++i) {
            nums1[m + i] = nums2[i];
        }

        // Now sort the entire nums1 array
        sort(nums1.begin(), nums1.end());
    }
};

File: MergeSortedArray/MergeSortedArray.py
class Solution:
    def merge(self, nums1, m, nums2, n):
        p1, p2, p = m - 1, n - 1, m + n - 1

        while p1 >= 0 and p2 >= 0:
            if nums1[p1] > nums2[p2]:
                nums1[p] = nums1[p1]
                p1 -= 1
            else:
                nums1[p] = nums2[p2]
                p2 -= 1
            p -= 1

        # Copy any remaining elements from nums2
        nums1[:p2 + 1] = nums2[:p2 + 1]

# Example usage
solution = Solution()
example1 = solution.merge([1, 2, 3, 0, 0, 0], 3, [2, 5, 6], 3)
example2 = solution.merge([1], 1, [], 0)
example3 = solution.merge([0], 0, [1], 1)

example1, example2, example3
